### PR TITLE
Add media layouts

### PR DESCRIPTION
**Pull request by lunson**
Adds media layouts for training materials to be used in December training.

### CODE REVIEW

**Comment by lunson** [comment]
Will someone who is more script-savvy than I please make the following work somehow?

In the `media.html` layouts file, please have button class `media-option-active` only apply to the page that the user is currently on. Right now, I only have it applying to the Docs page.

Thanks!

**Comment by hflynn** [comment]
Staged at https://snoopycrimecop.github.io/www.openmicroscopy.org/media/

**Comment by hflynn** [comment]
N.B. trigger for re-building snoopy site is https://ci.openmicroscopy.org/view/Mgmt/job/WWW-merge/

**Comment by waxenegger** [comment]
@lunson and @hflynn 

ok...this is  somewhat frustrating business not because the task in itself is so difficult with plain java script but this would be I believe the first custom js in the project and I was hoping to be able to use the framework, in specific something similar to what's shown here: https://zurb.com/university/lessons/advanced-panini-tips-and-tricks

I had the following: https://github.com/waxenegger/www.openmicroscopy.org/commit/3cde620cba8040730cb2790d15941421f9cb12ca

But nope. Trying it locally with jekyll would not work. The conditional I put there is just ignored, the page.dir variable is there (I left the output in the <p>). Not sure what's wrong, for all I could gather panini should be part of foundation 6 but no error, no nothing. It just ignores the conditional.

Feel free to pursue this idea above, if you get tired of it here is a java script solution: https://github.com/waxenegger/www.openmicroscopy.org/commit/5728f889e21db04ca08d33b0c9b24878ad3ecbcb

**Comment by lunson** [comment]
@waxenegger Thanks, Harald. Sorry it was frustrating. I'm fine with a custom js solution! I just didn't want to have to duplicate the navigation for all the pages.

**Comment by hflynn** [comment]
https://snoopycrimecop.github.io/www.openmicroscopy.org/media/ updated

**Comment by hflynn** [comment]
Presumably we intend to host the materials on the downloads site rather than copy everything to this repo @jburel ?

Also, looking at the materials listed for prep, I don't think we need an images section, do we?

**Comment by hflynn** [comment]
We have an example .pptx to use on https://downloads.openmicroscopy.org/presentations/2017/Cambridge-Training/ now

File: _layouts/media.html [hunk 1]
```diff
@@ -0,0 +1,24 @@
+---
+layout: default
+title: Media
+---          
+        <div class="callout large primary" id="bg-image-media" style="margin-bottom: 0rem;">
+            <div class="row column text-center">
+                <h1>Media</h1>
+                <p>[ Blurb about the December Training material (ppt, pdf, word docs, templates and videos) ]</p>
```
Comment (hflynn) [inline _layouts/media.html]: "These materials are available for you to use and adapt for your own training purposes"

**Comment by lunson** [comment]
@hflynn I've added the .pptx file to the slides section. Do you think we need blurbs for the sections? If not, I can delete.

**Comment by hflynn** [comment]
@lunson I'd say we don't need them personally, happy for you to go ahead and delete them.

**Comment by hflynn** [comment]
https://snoopycrimecop.github.io/www.openmicroscopy.org/media/slides/ updated @jburel

**Comment by jburel** [comment]
we also have a reveal presentation for Cambridge http://downloads.openmicroscopy.org/presentations/2017/Cambridge-Training/Image-Processing-OMERO

**Comment by jburel** [comment]
FYI the presentation available as ppx is also available as pdf.

**Comment by lunson** [comment]
@jburel Do you want the PDF to go under Slides, Docs, or both?

**Comment by jburel** [comment]
Slides please

File: media/slides/index.html [hunk 2]
```diff
@@ -28,6 +28,14 @@ <h2>Slides</h2>
                 </span>
                 </a>
                 <p class="text-center media-block-title">OMERO Users Training Day, December 2017 (PDF)</p>
+            </div>
+            <div class="media-block column image-hover-wrapper">
+                <a href="http://downloads.openmicroscopy.org/presentations/2017/Cambridge-Training/Image-Processing-OMERO/#/"><img src="{{ site.baseurl }}/media/slides/thumbnails/2017-12-image-processing-with-omero.png" alt="Image Processing with OMERO">
+                <span class="image-hover-wrapper-reveal">
+                    <p>Download slides<br><i class="fa fa-download" aria-hidden="true"></i></p>
```
Comment (hflynn) [inline media/slides/index.html]: Can you change this to 'View slides' with a different icon? Download looks odd when it's actually a hyperlink

File: media/slides/index.html [hunk 3]
```diff
@@ -20,6 +20,14 @@ <h2>Slides</h2>
                 </span>
                 </a>
                 <p class="text-center media-block-title">OMERO Users Training Day, December 2017</p>
+            </div>
+            <div class="media-block column image-hover-wrapper">
+                <a href="https://downloads.openmicroscopy.org/presentations/2017/Cambridge-Training/Day1-Cambridge2017-Introduction.pdf"><img src="{{ site.baseurl }}/media/slides/thumbnails/2017-12-cambridge-training-pdf.png" alt="December 2017 Cambridge Training">
+                <span class="image-hover-wrapper-reveal">
+                    <p>Download slides (PDF)<br><i class="fa fa-download" aria-hidden="true"></i></p>
```
Comment (hflynn) [inline media/slides/index.html]: Maybe just remove the 'slides' altogether in favour of 'Download PDF'?

**Comment by hflynn** [comment]
Looks good, thanks @lunson 

@jburel is the plan to convert the gdoc workflows to pdfs and make them available too? Trying to work out what else we need to do here before it's good to merge - remove the duplicate videos, add some content under docs, do we want to keep an images section for screenshots?

**Comment by hflynn** [comment]
Also not sure how this fits in with https://docs.google.com/document/d/1tYT1J7Z7KWXq6zKWL8px2lI8KID6AUsl4sW5hlxJmLY/edit @jburel

**Comment by jburel** [comment]
The g.doc I have started is mainly to gather all we had to do to prep for the sessions
The goal being to have a new repo "training" repo so this will give us an idea of what we need for that repo

The g.doc we have prepared will need to be converted into pdf (we had to do that in order to prepare the paper copy). There is no reason not to make them available when we have the repo in place so it will be similar to the links you have under slides. They will probably being under docs.
For the images I could see a "figure" as a screenshot and it could like to data resources
For example for cambridge we used the P-TRE images under http://downloads.openmicroscopy.org/images/DV/ 
So we could promote both figure and the resources repo

What you have currently in place is probably enough 
Thanks

**Comment by hflynn** [comment]
Can you put the training workflow pdfs on downloads then please @jburel 

We maybe need a wider discussion about whether we want screenshots to live on this repo or elsewhere and linked here

**Comment by hflynn** [comment]
https://snoopycrimecop.github.io/www.openmicroscopy.org/media/slides/ updated

**Comment by jburel** [comment]
pdf uploaded, discussed with @hflynn

**Comment by hflynn** [comment]
@lunson ok, so the example docs are the two pdfs on https://downloads.openmicroscopy.org/presentations/2017/Cambridge-Training/

The blurbs are as follows:

Day 1 - example workflows for managing images; viewing, analyzing and publishing images using OMERO plugins; filtering, searching and analyzing using Fiji features; and publishing with OMERO.figure.

Day 2 - example workflows for image processing with OMERO, client and server-side, including scripting, Jupyter notebooks and CellProfiler.

Once these are in place, can you remove the filler content and duplicated videos, and the 'images' tab (not sure whether we'll want this in future or not but we don't have any content for it now and need to have a wider discussion about where we want to host screenshots). Thanks!

**Comment by lunson** [comment]
@hflynn I called the blurbs "description"; let me know if there's a better word. "Preview"? "Intro"?

**Comment by hflynn** [comment]
Looks great to me @lunson thanks very much

**Comment by hflynn** [comment]
@jburel @joshmoore @sbesson can we get sign off on this as ready to merge now - the content is staged at https://snoopycrimecop.github.io/www.openmicroscopy.org/media/ and I'll ping Jason on the Trello card

**Comment by jburel** [comment]
Sorry one extra one done during the training prep. This could be added under slides
http://downloads.openmicroscopy.org/presentations/2017/SMG-symposium-workshop/#/

**Comment by jburel** [comment]
I will check with @eleanorwilliams about the recent IDR one. This will be another good example

**Comment by hflynn** [comment]
Title for the new presentation Jm linked to "OMERO introductory workshop slides" @lunson and this one will be another 'view slides' style item.

**Comment by jburel** [comment]
We might need to rename the original pdf. They cover a lot so very difficult to capture in the name itself

File: media/slides/index.html [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+                </a>
+                <p class="text-center media-block-title">OMERO Users Training Day,<br>December 2017</p>
+            </div>
+            <div class="media-block column image-hover-wrapper">
+                <a href="https://downloads.openmicroscopy.org/presentations/2017/Cambridge-Training/Day1-Cambridge2017-Introduction.pdf"><img src="{{ site.baseurl }}/media/slides/thumbnails/2017-12-cambridge-training-pdf.png" alt="December 2017 Cambridge Training">
+                <span class="image-hover-wrapper-reveal">
+                    <p>Download PDF<br><i class="fa fa-download" aria-hidden="true"></i></p>
+                </span>
+                </a>
+                <p class="text-center media-block-title">OMERO Users Training Day,<br>December 2017 (PDF)</p>
```
Comment (sbesson) [inline media/slides/index.html]: What's the contract here in terms of presentation format? The PDF format is specified but others are not.

File: media/index.html [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        <thead>
+            <tr>
+                <th>File Name</th>
+                <th width="250">Last modified</th>
+                <th width="150">Size</th>
+            </tr>
+        </thead>
+        <tbody>
+            <tr>
+                <td><i class="fa fa-file-pdf-o"></i><a href="https://downloads.openmicroscopy.org/presentations/2017/Cambridge-Training/walkthrough_day_1.pdf">walkthrough_day_1.pdf</a> <span data-tooltip aria-haspopup="true" class="has-tip top" data-disable-hover="false" tabindex="2" title="Day 1 - example workflows for managing images; viewing, analyzing and publishing images using OMERO plugins; filtering, searching and analyzing using Fiji features; and publishing with OMERO.figure.">(description)</span></td>
```
Comment (sbesson) [inline media/index.html]: If we are listing the filenames only, my main concern are the generic names and the confusion with future entries. Two options would be:
- prefix each presentation with the date/event but this is long creates some redundancy in the downloads URL
- group items by categories e.g. `2017 Cambridge Training`. Has such a layout been discussed and agreed?

File: media/slides/index.html (hunk 4 above)
Comment (hflynn) [reply media/slides/index.html]: Ah, we should probably have the other slides labelled as .pptx
Don't know about the Reveal ones - given that they open in the browser, listing a format many not not be familiar with might discourage them from clicking through? Maybe we should use 'web-based' instead?

Comment (mtbc) [reply media/slides/index.html]: :+1: for being vague about the exact web-based technology being used!

File: media/index.html (hunk 5 above)
Comment (lunson) [reply media/index.html]: I vote for and pushed the latter for the reasons you listed in the former.

File: media/slides/index.html (hunk 4 above)
Comment (lunson) [reply media/slides/index.html]: Adding "web-based" makes the file names longer and may force them into 3 lines so I went with some simple iconography. Hope that's okay.

Comment (hflynn) [reply media/slides/index.html]: Looks okay to me!

**Comment by hflynn** [comment]
https://snoopycrimecop.github.io/www.openmicroscopy.org/media/ updated

**Comment by hflynn** [comment]
@jburel should I merge this and then we can handle file renaming and maybe adding some more videos on another PR? Not expecting to need to update the live website in the next few days anyway so we needn't release it before renaming

**Comment by joshmoore** [comment]
Trying to catch up with emails etc, but if there's not an agreement on this page I wouldn't leave `master` in an unreleasable state.

**Comment by jburel** [comment]
discussion required so we are not ready yet to merge it

**Comment by hflynn** [comment]
Chatted with JM today, this is basically on hold until we have the new repo sorted and a better idea of what content we want to host here (probably January sometime).

One thing that i think is clear though @lunson is that we'll want a block of content below the header but above the 'docs/slides/videos' split where we can add extra text and links. Don't know if you want to do some styling for that now or wait 'til we have a better idea of what content we'll want to put there.

**Comment by lunson** [comment]
@hflynn Sounds good. Content first, so I know what to design for. Just ping me when it's ready in the new year!

**Comment by hflynn** [comment]
@lunson this will need master merging into it and the conflict fixed before we can include it again. No rush though, seems like we have other priorities just now.

**Comment by hflynn** [comment]
I'm going to close this, fix up the branch and open a fresh PR to continue this work for the users meeting

**Pull request closed without merging**